### DISCUSSION TITLE
external fontlist

### DISCUSSION THREAD

**Discussion by tbmuc** [ImageMagick/ImageMagick#3509] [Help]
Hello,

I have some bat-files which uses the same list of fonts. The actual font is selected by a bat parameter. In order to simplify maintainace, i would like to put the font selection into an extra bat file and call it by

Call fontlist.bat %1

Fontlist.bat consist of jumps like this
----------
if NOT %1  == "Schrift 1" goto :F2

set FONTFILE=Hey August.ttf
goto :ok

:F2 
-----------

The problem is, that it does not be possible to call a bat file which contains goto-jumps. Any idea?  

IM-Version 7.x

**Comment by snibgo** [comment]
The question is about Windows BAT scripts, not ImageMagick.

Does %1 start and end with quotes? If not, then your "if" will never be true.

**Reply by tbmuc** [reply]
Yes, they do. The font script works fine, the issues come when calling it from the main script. An error occurs that goto's are not allowed inside

**Comment by snibgo** [comment]
I suspect that Fontlist.bat is more complex than you have shown.

**Reply by tbmuc** [reply]
[fontlist.txt](https://github.com/ImageMagick/ImageMagick/files/6265788/fontlist.txt)
Not really, just a sequence of that.
